UnityProject/Assets/Scripts/Systems/Construction/MachineFrame.cs

```diff
@@ -454,11 +454,15 @@ private void PartCheck(GameObject usedObject, HandApply interaction)
 			//Main logic for tallying up and moving parts to hidden pos
 			if (basicPartsUsed.ContainsKey(itemTrait) && usedObject.GetComponent<Stackable>() != null && usedObject.GetComponent<Stackable>().Amount >= needed) //if the itemTrait already exists, and its stackable and some of it is needed.
 			{
-				basicPartsUsed[itemTrait] = machinePartsList.amountOfThisPart;
+				var StackingItem = usedObject.GetComponent<Stackable>();
+				var oldAmount = StackingItem.Amount;
+				var addNew = StackingItem.ServerRemoveOne();
+				addNew.GetComponent<Stackable>().ServerSetAmount(needed);
 
-				Inventory.ServerDrop(interaction.HandSlot);
+				StackingItem.ServerSetAmount(oldAmount -needed );
```

Comment:
why set the stack instead of dropping it?

Reply:
so, this is if you got too much to fit in the machine, Your removing some of that and putting it in the machine